src/bm_sim/checksums.cpp

```diff
@@ -149,7 +149,12 @@ CalcBasedChecksum::update_(Packet *pkt) const {
 bool
 CalcBasedChecksum::verify_(const Packet &pkt) const {
   const uint64_t cksum = calculation->output(pkt);
+  std::stringstream ss, ess;
+  ss << std::hex << cksum;
   const auto &f_cksum = pkt.get_phv()->get_field(header_id, field_offset);
+  ess << std::hex << f_cksum.get<uint64_t>();
+  BMLOG_DEBUG_PKT(pkt, "Checksum '{}': computed {} : in pkt {}",
+                  get_name(), "0x"+ ss.str(), "0x"+ ess.str());
```

Comment:
you can use `ss << "0x" << std::setfill('0') << std::setw(8) << std::hex << <value>`